From 9770ef433c14f736c8bf8aa41bd28b4861a93617 Mon Sep 17 00:00:00 2001
From: Vitor Soares <vitor.soares@toradex.com>
Date: Mon, 1 Jul 2024 13:43:02 +0100
Subject: [PATCH] arm64: dts: imx8mp: Fix VPU PGC power-domain parents

On iMX8M Plus QuadLite (VPU-less SoC), the dependency between VPU power
domains lead to a deferred probe error during boot:
[   17.140195] imx-pgc imx-pgc-domain.8: failed to command PGC
[   17.147183] platform imx-pgc-domain.11: deferred probe pending: (reason unknown)
[   17.147200] platform imx-pgc-domain.12: deferred probe pending: (reason unknown)
[   17.147207] platform imx-pgc-domain.13: deferred probe pending: (reason unknown)

This is incorrect and should be the VPU blk-ctrl controlling these power
domains, which is already doing it.

After removing the `power-domain` property from the VPU PGC nodes, both
iMX8M Plus w/ and w/out VPU boot correctly. However, it breaks the
suspend/resume functionality. A fix for this is pending, see Links.

Upstream-Status: Submitted [https://lore.kernel.org/all/20240701124302.16520-1-ivitro@gmail.com/]
Cc: stable@vger.kernel.org
Fixes: df680992dd62 ("arm64: dts: imx8mp: add vpu pgc nodes")
Link: https://lore.kernel.org/all/fcd6acc268b8642371cf289149b2b1c3e90c7f45.camel@pengutronix.de/
Link: https://lore.kernel.org/all/20240418155151.355133-1-ivitro@gmail.com/
Suggested-by: Lucas Stach <l.stach@pengutronix.de>
Signed-off-by: Vitor Soares <vitor.soares@toradex.com>
---
 arch/arm64/boot/dts/freescale/imx8mp.dtsi | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp.dtsi b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
index e0d3b8cba221..e69eb232562c 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
@@ -879,14 +879,12 @@ pgc_mediamix: power-domain@10 {
 
 					pgc_vpu_g1: power-domain@11 {
 						#power-domain-cells = <0>;
-						power-domains = <&pgc_vpumix>;
 						reg = <IMX8MP_POWER_DOMAIN_VPU_G1>;
 						clocks = <&clk IMX8MP_CLK_VPU_G1_ROOT>;
 					};
 
 					pgc_vpu_g2: power-domain@12 {
 						#power-domain-cells = <0>;
-						power-domains = <&pgc_vpumix>;
 						reg = <IMX8MP_POWER_DOMAIN_VPU_G2>;
 						clocks = <&clk IMX8MP_CLK_VPU_G2_ROOT>;
 
@@ -894,7 +892,6 @@ pgc_vpu_g2: power-domain@12 {
 
 					pgc_vpu_vc8000e: power-domain@13 {
 						#power-domain-cells = <0>;
-						power-domains = <&pgc_vpumix>;
 						reg = <IMX8MP_POWER_DOMAIN_VPU_VC8000E>;
 						clocks = <&clk IMX8MP_CLK_VPU_VC8KE_ROOT>;
 					};
-- 
2.45.2

